decorators.go

```diff
@@ -637,7 +637,7 @@ func (CommandWithExecuteDecorator) Decorate(_ *Ecdysis, cmd *cobra.Command, c Co
 			}
 		}
 
-		ctx := contextWithCobraCommand(cmd.Context(), cmd)
+		ctx := ContextWithCobraCommand(cmd.Context(), cmd)
```

Comment:
Out of curiosity, why would Conduit need to set a Cobra command? I was under impression that ecdysis should take care of the Cobra stuff?

Reply:
@hariso This is to be consistent with the `Execute` decorator we currently have in ecdysis. The new `ExecuteWithClient` misses the ability to have access to the cobra command if we need to. 

For context, this was needed for situations like this one in `conduit` (root) https://github.com/ConduitIO/conduit/blob/b32c7c0826cb1688ed303767ff79abbf6dbbda83/cmd/conduit/root/root.go#L54-L56.

Fwiw, this was me trying to be consistent with the other method, but what I really need from this PR is the `ParseConfig` method now. Renaming PR title to make that clearer.